cli/automation/bitbucket-pipelines/index.md

```diff
@@ -0,0 +1,219 @@
+---
+title: How to use Terramate to automate and orchestrate Terraform in BitBucket Pipelines
+description: Learn how to use Terramate to configure custom GitOps workflows to automate and orchestrate Terraform and OpenTofu in BitBucket Pipelines.
+---
+# Automating Terramate in BitBucket Pipelines
+
+Bitbucket Pipelines add continuous integration to Bitbucket repositories to automate your software builds, tests, and deployments. Automating Terraform with CI/CD enforces configuration best practices, promotes collaboration, and automates the Terraform workflow.
+
+Terramate integrates seamlessly with Bitbucket Pipelines to automate and orchestrate IaC tools like Terraform and OpenTofu.
+
+## Terramate Blueprints
+
+This page explains the workflow setup and authentication flows common in the following workflows.
+
+To jump directly to the Blueprints, follow the links below:
+
+- [Pull Request Preview Workflow Blueprints](./preview-workflow.md)
+- [Deployment Workflow Blueprints](./deployment-workflow.md)
+<!-- - [Drift Check Workflow Blueprints](./drift-check-workflow.md) -->
+
+The pipelines in these examples rely on a few Shell scripts to run different parts of the workflow. All these scripts are created inside a folder called `bitbucket-scripts`. The following sections describe how each of these scripts work.
+
+## Installing the necessary packages
+
+The workflows in these examples use the `google/cloud-sdk` docker image based on a minimal image and include the `gcloud` packages.
+
+In addition to installing Terraform and Terramate, you need other packages for the workflows to work. The installation is done using `asdf` which requires another file called `.tool-versions`.
+
+Create the following at `bitbucket-scripts/install.sh`
+```bash
+#!/bin/bash
+
+apt-get install -y unzip jq
+git clone https://github.com/asdf-vm/asdf.git ~/.asdf --branch v0.14.0 && . ~/.asdf/asdf.sh
+asdf plugin add terraform
+asdf plugin add terramate
+asdf install
+```
+
+And `.tool-versions`
+```
+terraform 1.9.3
+terramate 0.9.3
+```
+
+## Authenticating to Google Cloud
+
+This script requires 2 inputs: Worklaod Identity Provider (WIP) ID and the Service Account email.
+
+Create the following file at `bitbucket-scripts/gcp-oidc-auth.sh`
+
+```bash
+#!/bin/bash
+
+WORKLOAD_IDENTITY_PROVIDER=$1
+SERVICE_ACCOUNT=$2
+
+echo "${BITBUCKET_STEP_OIDC_TOKEN}" > /tmp/gcp_access_token.out
+gcloud iam workload-identity-pools create-cred-config "${WORKLOAD_IDENTITY_PROVIDER}" --credential-source-file=/tmp/gcp_access_token.out --service-account="${SERVICE_ACCOUNT}" --output-file=/tmp/sts-creds.json
+export GOOGLE_APPLICATION_CREDENTIALS=/tmp/sts-creds.json
+gcloud auth login --cred-file=/tmp/sts-creds.json
```

Comment:
I think we should add:
```
set -euo pipefail
```
to avoid unintended consequences if environment is not set or not all arguments is passed to the script.